commercial/app/model/commercial/Merchandise.scala

```diff
@@ -0,0 +1,370 @@
+}
+
+case class Member(username: String, gender: Gender, age: Int, profilePhoto: String, location: String) {
+
+  val profileId: Option[String] = profilePhoto match {
+    case Member.IdPattern(id) => Some(id)
+    case _ => None
+  }
+
+  val profileUrl: String = profileId.map(id => s"https://soulmates.theguardian.com/landing/$id")
+    .getOrElse("http://soulmates.theguardian.com/")
+
+}
+
+case class MemberPair(member1: Member, member2: Member) extends Merchandise
+
+sealed case class Gender(name: String) {
+  override def toString = name
+}
+
+case class Job(id: Int,
+               title: String,
+               shortDescription: String,
+               locationDescription: Option[String],
+               recruiterName: String,
+               recruiterPageUrl: Option[String],
+               recruiterLogoUrl: String,
+               sectorIds: Seq[Int],
+               salaryDescription: String,
+               keywordIdSuffixes: Seq[String] = Nil) extends Merchandise {
+
+  val shortSalaryDescription = StringUtils.abbreviate(salaryDescription, 25).replace("...", "…")
+
+  def listingUrl = s"http://jobs.theguardian.com/job/$id"
+
+  val industries: Seq[String] =
+    Industries.sectorIdIndustryMap.filter { case (sectorId, name) => sectorIds.contains(sectorId)}.values.toSeq
+
+  val mainIndustry: Option[String] = industries.headOption
+}
+
+object Merchandise {
+  val writes: Writes[Merchandise] = new Writes[Merchandise] {
+    def writes(m: Merchandise) = m match {
+      case b: Book => Json.toJson(b)
+      case j: Job  => Json.toJson(j)
+      case m: Masterclass => Json.toJson(m)
+      case m: Member => Json.toJson(m)
+      case p: MemberPair => Json.obj(
+        "member1" -> Json.toJson(p.member1),
+        "member2" -> Json.toJson(p.member2)
+      )
+      case t: TravelOffer => Json.toJson(t)
+      case l: LiveEvent => Json.toJson(l)
+    }
+  }
+}
+
+object Book {
+
+  private val authorReads = {
+    ((JsPath \ "author_firstname").readNullable[String] and
+      (JsPath \ "author_lastname").readNullable[String])
+      .tupled.map { case (optFirstName, optLastName) =>
+      for {
+        firstName <- optFirstName
+        lastName <- optLastName
+      } yield s"$firstName $lastName"
+    }
+  }
+
+  private def stringOrDoubleAsDouble(value: String): Reads[Option[Double]] = {
+    val path = JsPath \ value
+    path.readNullable[Double] orElse path.readNullable[String].map(_.map(_.toDouble))
+  }
+
+  implicit val bookReads: Reads[Book] = (
+    (JsPath \ "name").read[String] and
+      authorReads and
+      (JsPath \ "isbn").read[String] and
+      stringOrDoubleAsDouble("regular_price_with_tax") and
+      stringOrDoubleAsDouble("final_price_with_tax") and
+      (JsPath \ "description").readNullable[String] and
+      (JsPath \ "images")(0).readNullable[String] and
+      (JsPath \ "product_url").readNullable[String] and
+      (JsPath \ "guardian_bestseller_rank").readNullable[String].map(_.map(_.toDouble.toInt)) and
```

Comment:
Appreciate you didn't write this so maybe I can pick this up in a different PR, but it looks  a bit funny reading as a `String`, mapping it twice to a `Double` and then to an `Int`... Are we aiming to round down the ranking or are we just going really round the houses to read an `Int`? Might be worth checking the feed to see what is going on.